### PR TITLE
add metadata to relevant schedule methods

### PR DESCRIPTION
**Pull request by KPostOffice**
## Related Issues and Dependencies

thoth-station/kebechet#584

## This introduces a breaking change

- [ ] Yes
- [x] No

## This Pull Request implements

Add metadata to both kebechet-run-url which is triggered by `kebechet-admin` in workflow helpers as well as `adviser` so that the metadata can be passed when using advise and provenance checkers.  The metadata will be used by kebechet to construct PR and issue bodies. In the case of the advise and provenance manager, the metadata will be sent to adviser and provenance workflows and then be received by kebechet-run-results.

### CODE REVIEW

**Review by pacospace — COMMENTED** [review]
only one question, otherwise lgtm!

File: thoth/common/openshift.py [hunk 1]
```diff
@@ -1588,6 +1593,7 @@ def schedule_kebechet_run_url_workflow(
         repo_url: str,
         service_name: str = "github",
         *,
+        kebechet_metadata: Dict[str, Any] = {},
```
Comment (pacospace) [inline thoth/common/openshift.py:1596]: why not `Optional[Dict[str, Any]] = None`?

File: thoth/common/openshift.py [hunk 2]
```diff
@@ -1452,7 +1455,9 @@ def schedule_provenance_checker(
         template_parameters = {
             "THOTH_ADVISER_REQUIREMENTS": requirements,
             "THOTH_ADVISER_REQUIREMENTS_LOCKED": requirements_locked,
-            "THOTH_ADVISER_METADATA": json.dumps({"origin": origin}),
+            "THOTH_ADVISER_METADATA": json.dumps(
+                {"origin": origin, "kebechet_metadata": kebechet_metadata}
```
Comment (fridex) [inline thoth/common/openshift.py:1459]: We should sync on this one. I opened https://github.com/thoth-station/adviser/pull/1625 for this and the semantics behind that is:

Keep metadata generic, so anything can be passed in the JSON. Any specific entries like `stack_info` or justification propagated to the adviser output should go to `thoth.adviser` key so we can possibly expand the metadata. WDYT?

Comment (KPostOffice) [reply thoth/common/openshift.py:1459]: I'm not sure that I understand what you mean.

Comment (KPostOffice) [reply thoth/common/openshift.py:1459]: @fridex Are you saying get rid of https://github.com/thoth-station/common/pull/1103/files#diff-00668a241bbe4ebb4fe69571ac2634fba90bd935bfb267345ec0c946bfe6bdccL1338-L1343 and put it all under the generic argument `metadata` so we can easily add new metadata?

**Review by fridex — APPROVED** [review]
Thanks :+1:  I will open a follow-up PR to review/sync.

**Comment by fridex** [comment]
@KPostOffice PTAL on conflicts.

**Pull request merged**